### ISSUE TITLE
Drift_dev code gen incorrect assumption about type nullability

### ISSUE THREAD

**Issue by tcd93** [simolus3/moor#2692]
**REPRODUCE STEPS**
To reproduce issue on repo's example app (with drift_dev 2.13.0)

1. _sql.drift_ 
```sql
CREATE TABLE money(amount REAL NOT NULL);
sumTracker: SELECT sum(amount) FROM money;
```

2. _database_test.dart_
```dart
  test('can watch money', () async {
    final stream = database.sumTracker().watchSingleOrNull();
    stream.listen((value) {
      print(value);
    }, onError: (Object error, StackTrace stack) {
      print('ERROR: $error');
    });

    await database
        .into(database.money)
        .insert(MoneyCompanion.insert(amount: 100.0));
    await Future.delayed(const Duration(milliseconds: 500));
    await (database.delete(database.money)).go();
    await Future.delayed(const Duration(milliseconds: 500));
  });
```

I expected the printed values be: 
```
null
100
null
```
But I got errors instead:
```
ERROR: type 'Null' is not a subtype of type 'double' in type cast
100.0
ERROR: type 'Null' is not a subtype of type 'double' in type cast
```

The issue is that `sumTracker` return Object type is `Seletable<double>`, which is not always the case, when the table is empty, there's no rows to be selected hence null returned

**Workaround**: make column `amount` nullable, `sumTracker` now returns the correct `Selectable<double?>`

**Comment by simolus3** [comment]
Thanks for the report, I've fixed this in d70de95382c7cd63fe66ef7008301ec2dc00566d.

> **Workaround**: make column `amount` nullable, `sumTracker` now returns the correct `Selectable<double?>`

You can also use `sumTracker: SELECT nullif(sum(amount), null) FROM money;` as a workaround. It doesn't change the query, but drift will take that as a hint to make the column nullable.

**Comment by tcd93** [comment]
Just a reminder that this issue can happen with other aggregators like MIN MAX AVG, maybe GROUP_CONCAT too.

Edit: And thank you for the fast fix 👍
